### DISCUSSION TITLE
Increase player speed

### DISCUSSION THREAD

**Discussion by Squander12** [MCPHackers/RetroMCP-Java#109] [Q&A]
I added boots that increase walking speed, in theory everything works, but when I enter the server and try to walk, the server moves me back and gets the following information:
Got position 1.5275054012421543, 79.0, 2.406983900200199
Expected 1.7577250298625335, 79.0, 2.5257922519494524

Client and server code in EntityPlayer
@Override
	public void moveEntity(double x, double y, double z) {
		ItemStack boots = inventory.armorInventory[0];
		if(boots != null && boots.getItem() == Item.windBoots) {
			x *= 1.9f;
			z *= 1.9f;
		}
		
		super.moveEntity(x, y, z);
	}

**Comment by PhoenixVX** [comment]
You need to apply the code edits to the server too if you want to make the client & server not desync. Aside from that, if it's still too fast, you might need to disable the speed check in the server code or modify the values it uses.

**Reply by Squander12** [reply]
### The code on the server and client sides is the same, it seems to me that the Minecraft beta server has some anti-cheat that teleports me to the previous place.

**Reply by PhoenixVX** [reply]
Yeah, there is a simple one. Ctrl-F for `"Expected "` and modify the values used to determine if a player is moving too fast.

**Reply by Squander12** [reply]
I found this code in the NetServer Handler class, but I don't know whether to turn it off or change the values.
`playerZ = zPlayer - this.playerEntity.posZ;
			distance = deltaStance * deltaStance + playerY * playerY + playerZ * playerZ;
			boolean movedWrongly = false;
			
			if(distance > 1.0D / 16.0D && !this.playerEntity.isSleeping()) {
				movedWrongly = true;
				logger.warning(this.playerEntity.username + " moved wrongly!");
				System.out.println("Got position " + xPlayer + ", " + yPlayer + ", " + zPlayer);
				System.out.println("Expected " + this.playerEntity.posX + ", " + this.playerEntity.posY + ", " + this.playerEntity.posZ);
			}

			this.playerEntity.setPositionAndRotation(xPlayer, yPlayer, zPlayer, yaw, pitch);
			boolean noCollisionsAfterMove = serverWorld.getCollidingBoundingBoxes(this.playerEntity, this.playerEntity.boundingBox.copy().getInsetBoundingBox((double)d, (double)d, (double)d)).size() == 0;
			
			if(noCollisions && (movedWrongly || !noCollisionsAfterMove) && !this.playerEntity.isSleeping()) {
				this.teleportTo(this.lastPosX, this.lastPosY, this.lastPosZ, yaw, pitch);
				return;
			}`

**Reply by PhoenixVX** [reply]
You can change the `distance > 1.0D / 16.0D` check. The maximum distance a player can move in a single tick according to this code is 0.0625. You'll want to modify it by however much the speed increase is (e.g. 1.5x).

**Reply by Squander12** [reply]
Ok, thanks.
